cmd/cluster/core/create.go

```diff
@@ -531,6 +527,17 @@ func GetAPIServerAddressByNode(ctx context.Context, l logr.Logger) (string, erro
 }
 
 func Validate(ctx context.Context, opts *CreateOptions) error {
+	if opts.Wait && opts.NodePoolReplicas < 1 {
+		return errors.New("--wait requires --node-pool-replicas > 0")
+	}
+
+	// Validate HostedCluster name follows RFC1123 standard
```

Comment:
the next validation at L542, checks if a HostedCluster resource with this name exist in the cluster, but If the name is invalid in the first place, we don't have to check that because `client.Get` call will fail anyway.